tensorboard/compat/proto/update.sh

```diff
@@ -16,31 +16,26 @@
 
 set -e
 
-# To find other protos you need run something like
-#
-#     find ../tensorflow/ -type f  -name 'event.proto'
-#
-
-# Copy all protos temporarily
-cp ../tensorflow/tensorflow/core/framework/*.proto tensorboard/compat/proto/
-cp ../tensorflow/tensorflow/core/protobuf/*.proto tensorboard/compat/proto/
-cp ../tensorflow/tensorflow/core/profiler/*.proto tensorboard/compat/proto/
-cp ../tensorflow/tensorflow/core/util/*.proto tensorboard/compat/proto/
-cp ../tensorflow/tensorflow/python/framework/*.proto tensorboard/compat/proto/
+if [ $# -ne 1 ]; then
+  echo "usage: $0 <tensorflow-root-dir>" >&2;
+  exit 1;
+fi
 
-# Only keep those that update
-# TODO: Might want to do this more selectively
-git clean -fdx
+rsync --existing "$1"/tensorflow/core/framework/*.proto tensorboard/compat/proto/
+rsync --existing "$1"/tensorflow/core/protobuf/*.proto tensorboard/compat/proto/
+rsync --existing "$1"/tensorflow/core/profiler/*.proto tensorboard/compat/proto/
+rsync --existing "$1"/tensorflow/core/util/*.proto tensorboard/compat/proto/
+rsync --existing "$1"/tensorflow/python/framework/*.proto tensorboard/compat/proto/
 
-# Replace paths internally
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|tensorflow/core/framework|tensorboard/compat/proto|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|tensorflow/core/protobuf|tensorboard/compat/proto|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|tensorflow/core/profiler|tensorboard/compat/proto|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|tensorflow/core/util|tensorboard/compat/proto|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|tensorflow/python/framework|tensorboard/compat/proto|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|package tensorflow.tfprof;|package tensorboard;|g' {} +
-find tensorboard/compat/proto/ -type f  -name '*.proto' -exec sed -i '' 's|package tensorflow;|package tensorboard;|g' {} +
+# Rewrite file paths and package names.
+find tensorboard/compat/proto/ -type f  -name '*.proto' -exec perl -pi \
```

Comment:
> `sed -i ''` is macOS-only

But `sed -i -e ...` works on both Linux and macOS. Mind using that
instead?

Reply:
It's not working on my macOS:

```bash
$ uname
Darwin
$ echo "foo" > /tmp/foo
$ sed -i -e 's/foo/bar/' /tmp/foo
$ cat /tmp/foo  # good
bar
$ ls /tmp/foo*  # less good
/tmp/foo	/tmp/foo-e
```

See also https://stackoverflow.com/a/5694430

Reply:
Ugh. That’s unfortunate. TIL. Thanks.